cpp/src/fil/fil.cu

```diff
@@ -586,20 +586,35 @@ void tl2fil_common(forest_params_t* params, const tl::Model& model,
   // fill in forest-dependent params
   params->depth = max_depth(model);  // also checks for cycles
 
+  const tl::ModelParam& param = model.param;
+
   // assuming either all leaves use the .leaf_vector() or all leaves use .leaf_value()
   size_t leaf_vec_size = tl_leaf_vector_size(model);
   if (leaf_vec_size > 0) {
     ASSERT(leaf_vec_size == model.num_output_group,
            "treelite model inconsistent");
     params->num_classes = leaf_vec_size;
     params->leaf_payload_type = leaf_value_t::INT_CLASS_LABEL;
+    ASSERT(
+      param.pred_transform != "identity",
+      "multiclass predict() only supported by choosing most likely label,\n"
+      "please use pred_transform == 'max_index' ('sigmoid' is still supported,"
+      " with 'max_index' implied)");
   } else {
     params->leaf_payload_type = leaf_value_t::FLOAT_SCALAR;
     params->num_classes = 0;  // ignored
+    if (param.pred_transform == "max_index" &&
```

Comment:
It might be a good idea to add a new field in Treelite that explicitly encodes whether the model is performing regression or classification, or multi-class classification. @Salonijain27 What do you think?

Reply:
That would be really nice to have!